include/data_management/data_source/file_data_source.h

```diff
@@ -196,7 +196,7 @@ class FileDataSource : public CsvDataSource<FeatureManager, SummaryStatisticsTyp
             return services::throwIfPossible(services::ErrorOnFileOpen);
         }
 #else
-        _file = fopen((char *)(fileName.c_str()), "r");
+        _file = std::fopen((char *)(fileName.c_str()), "r");
```

Comment:
If this worked previously, then we have "using namespace std" or "#include <stdio.h>" somewhere in the includes. But I can't find any of those. I am Ok with the change, but can you please explain how this worked previously?

Reply:
First of all, according to [iso cpp core guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-using-directive) we mustn't write `using namespace std` inside headers.
The second one, fopen without namespace is available for compatibility with C. We have included cstdio in the begging of the file. So we should use [std::fopen](https://en.cppreference.com/w/cpp/io/c/fopen)

Reply:
I do know that "using namespace std" is prohibited. But I didn't know that fopen exists in global namespace. Thanks for explanation!

Reply:
OK to have 'std::' here. The possible reason why it worked before is inclusion of stdio.h in another file that included here indirectly.